pkg/printers/internalversion/describe.go

```diff
@@ -2441,6 +2441,10 @@ func (dd *DeploymentDescriber) Describe(namespace, name string, describerSetting
 			}
 			w.Write(LEVEL_0, "NewReplicaSet:\t%s\n", printReplicaSetsByLabels(newRSs))
 		}
+		overlapWith := d.Annotations[deploymentutil.OverlapAnnotation]
+		if len(overlapWith) > 0 {
+			w.Write(LEVEL_0, "!!!WARNING!!! This deployment has overlapping label selector with deployment %q and won't behave as expected. Please fix it before continuing.\n", overlapWith)
```

Comment:
Why not remove it in `syncDeployment()` where we used to call [handleOverlap()](https://github.com/kubernetes/kubernetes/pull/42175/commits/92d75cbb2338c27a1404a556a2c3b92f52041941#diff-f88c99835d4b2acda532e36123761825L515)? It seems easier to prove that that will do the same thing as the old code would if the selectors no longer overlap.